sample-app/src/components/SearchBar.tsx

```diff
@@ -19,17 +20,20 @@ interface Props {
  */
 export default function SearchBar({ placeholder, isVertical, screenReaderInstructionsId }: Props) {
   const answersActions = useAnswersActions();
-  const query = useAnswersState(state => state.query.query);
-  const mapStateToAutocompleteResults: StateSelector<AutocompleteResult[] | undefined> = isVertical
-    ? state => state.vertical.autoComplete?.results
-    : state => state.universal.autoComplete?.results;
-  const autocompleteResults = useAnswersState(mapStateToAutocompleteResults) || [];
-  const isLoading = useAnswersState(state => state.vertical.searchLoading || state.universal.searchLoading);
-
-  function executeAutocomplete () {
+  const query = useAnswersState(state => state.query.input);
+  const [ autocompleteResults, setAutoCompleteResults ] = useState<AutocompleteResult[]>([]);
+  const isLoading = useAnswersState(state => state.searchStatus.isLoading);
+  const autocompleteNetworkIds = useRef({ latestRequest: 0, responseInState: 0 });
+  async function executeAutocomplete () {
```

Comment:
could we make a useAutocomplete hook? I was thinking of something like the below, it wouldn't go into the react bindings (at least not yet) but just live in some sample-app/src/hooks/ folder
```ts
function useAutocomplete(isVertical) {
  const autocompleteNetworkIds = useRef({ latestRequest: 0, responseInState: 0 });
  const [ autocompleteResults, setAutoCompleteResults ] = useState<AutocompleteResult[]>([]);
  async function executeAutocomplete () {
    // what you have
  }
  return [ autocompleteResults, executeAutocomplete ]
}
```

Reply:
I think it's ok to make it very specific to autocomplete for now, later on if we need we can make a more general hook that just does the network request race condition handling

Reply:
yeah, makes sense